### DISCUSSION TITLE
context in the irradiance calculation (sky view factor)

### DISCUSSION THREAD

**Discussion by DAltieri86** [pvlib/pvlib-python#1563] [Q&A]
hi all,
I've only been using pvlib for a few days now . 
I am wondering if there is a way to take context into account when calculating irradiance.
i'm referring to something that takes the sky view factor into account.

Thank you

**Comment by kurt-rhee** [comment]
Heya, 

I'm not entirely sure if I understand what context means, but as I understand it you want to know how much irradiance will hit a panel if it is tilted rather than on a flat surface.  I've included some pictures for reference.  If that is true I use pvlib's transpositions models to do that.  I included some example documentation below.  If you are interesting in another topic I could try to help point you in the right direction as well.

![image](https://user-images.githubusercontent.com/33131958/193272630-25fa88ed-0287-40d2-aa79-4429085a1bc2.png)
![image](https://user-images.githubusercontent.com/33131958/193272660-ca781eb5-1bd6-4773-9792-74b7e1e78785.png)

https://pvlib-python.readthedocs.io/en/stable/gallery/irradiance-transposition/plot_ghi_transposition.html

**Reply by DAltieri86** [reply]
Thank you very much for your response.
I apologize if i am not using the correct terminology; solar is not my field. 
I try to explain myself better. I am interested in the calculation of solar potential with reference to building facades. These are often subject to more or less significant shading due to the urban environment (buildings, trees, etc), especially on lower floors.
I wanted to find out if there is a possibility in PVlib to include any obstacles causing shading in the irradiance calculation. I hope this help.. Thanks again

**Comment by kurt-rhee** [comment]
Ah I understand now,

As I understand pvlib, there is no geo-spacial data held by the solar panel
array, and therefore no external object shading available (shading from
things that aren't other pv modules in the same array).

There are other tools in the industry such as PVSyst, PlantPredict and Daly
Energy which have external object shading models available.  If you are
looking to work with Python then PlantPredict or DalyEnergy is the way to
go since they both have an API and PlantPredict has an SDK.  In general
PlantPredict is free, but the API is behind a paywall.  DalyEnergy is a
startup that has very detailed modeling on terrain and diode based
submoduling, I am not sure what the cost for their API is.  Daly doesn't
have a website yet, but I can get you in touch with them if you are
interested.

https://plantpredict.com/

**Reply by kandersolar** [reply]
Correct, pvlib (currently) has no built-in ability to understand and model an external shading scene.  Of course it is all in python, so you could in principle perform these calculations yourself outside of pvlib, but this is a nontrivial task.  

As far as other software goes, I'd be remiss if I didn't mention SAM, which is free & open-source and has 3D shade scene functionality:
- SAM website: https://sam.nrel.gov/
- 3D scene documentation: https://samrepo.nrelcloud.org/help/index.html?3dshad-how-to.htm

**Reply by kurt-rhee** [reply]
^ I entirely agree that SAM would be a great tool for this.

**Reply by DAltieri86** [reply]
Thank you all for your kind and very accurate answers.
Just an additional detail. I would be interested not in a detailed calculation of the surrounding elements, but in a single variable from 0 (totally obscured) to 1 (totally free view) (e.g. sky vie factor) linked to the shading level that is used in the calculation to reduce the solar potential accordingly. The aim is to calibrate a metamodel that tries to predict the solar potential on the basis of certain characteristics such as the degree of shading. Anyway,thanks again for your valuable advice

**Reply by DAltieri86** [reply]
I will look in detail at your references

**Comment by kurt-rhee** [comment]
Heya,

Given that information, I would give pvmismatch a shot.  It won't give you
the energy for the whole system (dc / ac line losses, inverter performance
and clipping, transformer losses, etc), but it will give you the
performance of a given array.

https://sunpower.github.io/PVMismatch/



DAltieri86 wrote:


Thank you all for your kind and very accurate answers.
Just an additional detail. I would be interested not in a detailed
calculation of the surrounding elements, but in a single variable from 0
(totally obscured) to 1 (totally free view) (e.g. sky vie factor) linked to
the shading level that is used in the calculation to reduce the solar
potential accordingly. The aim is to calibrate a metamodel that tries to
predict the solar potential on the basis of certain characteristics such as
the degree of shading. Anyway,thanks again for your valuable advice

—
Reply to this email directly, view it on GitHub
<https://github.com/pvlib/pvlib-python/discussions/1563#discussioncomment-3772444>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/AH4Y3NSXSQH7ZBJC4ZL4MV3WA3TRBANCNFSM6AAAAAAQZXRI6U>
.
You are receiving this because you commented.Message ID:
***@***.***>
